File: Core/Architect.API.Core.DataFactory/_Generated/Labels.cs
using Architect.DataFactory;
using Architect.Utilities.Extensions;
using System;
using System.Collections.Generic;
using System.Data;
using DbType = Architect.DataFactory.Enumerations.DbType;

namespace Architect.API.Core.DataAccess.General
{
    /// <summary>
    /// Etiquetas
    /// </summary>
    public static partial class Labels
    {

        /// <summary>
        /// Crea un registro en la tabla Labels.
        /// </summary>
        /// <param name="labelsItem">Instancia de Labels</param>
        /// <param name="connection">Instancia de una conexión compartida</param>
        /// <returns>Cantidad de registros creados.</returns>
        public static int Create(Architect.API.Core.Contracts.General.Labels labelsItem, IDbConnection connection = null)
        {
            if (labelsItem.UpdateDate.IsEmpty())
            {
                labelsItem.UpdateDate = DateTime.Now;
            }
            return Database.Insert("INSERT INTO Labels (Id, CompanyId, EntityType, EntityId, LabelId, UpdateUserCode, UpdateDate) " +
                                                 "VALUES(:Id, :CompanyId, :EntityType, :EntityId, :LabelId, :UpdateUserCode, :UpdateDate)")
                            .AddParameter("Id", DbType.Decimal, 9, labelsItem.Id)
                            .AddParameter("CompanyId", DbType.Decimal, 5, labelsItem.CompanyId)
                            .AddParameter("EntityType", DbType.Decimal, 5, labelsItem.EntityType)
                            .AddParameter("EntityId", DbType.Decimal, 18, labelsItem.EntityId)
                            .AddParameter("LabelId", DbType.Decimal, 5, labelsItem.LabelId)
                            .AddParameter("UpdateUserCode", DbType.Decimal, 9, labelsItem.UpdateUserCode)
                            .AddParameter("UpdateDate", DbType.DateTime, 0, labelsItem.UpdateDate)
                            .Execute(connection, "Research");
        }

        /// <summary>
        /// Crea una lista de registros en la tabla Labels.
        /// </summary>
        /// <remarks>Complemento para procesamiento masivo</remarks>
        /// <param name="labelsItems">Lista de instancia de Labels</param>
        /// <param name="connection">Instancia de una conexión compartida</param>
        /// <returns>Lista con el resultado de la creación de cada instancia.</returns>
        public static List<int> Create(List<Architect.API.Core.Contracts.General.Labels> labelsItems, IDbConnection connection = null)
        {
            List<int> result = new List<int>();
            bool local = false;

            if (connection == null)
            {
                connection = Architect.DataFactory.Database.OpenConnection("Research");
                local = true;
            }
            foreach (Architect.API.Core.Contracts.General.Labels item in labelsItems)
            {
                result.Add(Create(item, connection));
            }
            if (local)
            {
                connection.Close();
            }
            return result;
        }

        /// <summary>
        /// Recupera un registro en la tabla Labels por medio de su clave primaria.
        /// </summary>
        /// <param name="id">Identificación única del registro.</param>
        /// <param name="companyId">Identificación de la compañia propietaria.</param>
        /// <param name="connection">Instancia de una conexión compartida</param>
        /// <returns>Instancia de Labels</returns>
        public static Architect.API.Core.Contracts.General.Labels Retrieve(int id, int companyId, IDbConnection connection = null)
        {
            Architect.API.Core.Contracts.General.Labels result = null;
            Database.Select("SELECT Id, Labels.CompanyId, EntityType, EntityId, LabelId, Labels.UpdateUserCode, um.FirstName || ' ' || um.LastName AS UpdateUserName, Labels.UpdateDate " +
                              "FROM Labels LEFT JOIN UserMember um ON um.UserId = Labels.UpdateUserCode " +
                             "WHERE Labels.Id=:Id AND Labels.CompanyId=:CompanyId")
                        .AddParameter("Id", DbType.Decimal, 9, id)
                        .AddParameter("CompanyId", DbType.Decimal, 5, companyId)
                        .Query(connection, "Research", new Action<System.Data.IDataReader>((reader) =>
                        {
                            result = DataReaderToLabels(reader);
                        }));
            return result;
        }

        /// <summary>
        /// Recupera una lista de registros en la tabla Labels.
        /// </summary>
        /// <param name="companyId">Identificación de la compañia propietaria.</param>
        /// <param name="filter">Filtro personalizado.</param>
        /// <param name="parameters">Lista de parametros para complemento.</param>
        /// <param name="connection">Instancia de una conexión compartida</param>
        /// <returns>Lista de instancias de Labels</returns>
        public static List<Architect.API.Core.Contracts.General.Labels> RetrieveAll(int companyId, string filter, List<DataFactory.Contracts.Parameter> parameters = null, IDbConnection connection = null)
        {
            List<Architect.API.Core.Contracts.General.Labels> result = new List<Architect.API.Core.Contracts.General.Labels>();
            Database.Select("SELECT Id, Labels.CompanyId, EntityType, EntityId, LabelId, Labels.UpdateUserCode, um.FirstName || ' ' || um.LastName AS UpdateUserName, Labels.UpdateDate " +
                              "FROM Labels LEFT JOIN UserMember um ON um.UserId = Labels.UpdateUserCode " +
                             "WHERE Labels.CompanyId=:CompanyId" + filter)
                        .AddParameter("CompanyId", DbType.Decimal, 5, companyId)
                        .AddParameter(parameters)
                        .Query(connection, "Research", new Action<System.Data.IDataReader>((reader) =>
                        {
                            result.Add(DataReaderToLabels(reader));
                        }));
            return result;
        }

        /// <summary>
        /// Recupera una lista de registros en la tabla Labels.
        /// </summary>
        /// <param name="companyId">Identificación de la compañia propietaria.</param>
        /// <param name="filter">Filtro personalizado.</param>
        /// <param name="beginIndex">Indice inicial para el paginado.</param>
        /// <param name="endIndex">Indice final para el paginado.</param>
        /// <param name="parameters">Lista de parametros para complemento.</param>
        /// <param name="connection">Instancia de una conexión compartida</param>
        /// <returns>Lista de instancias de Labels</returns>
        public static List<Architect.API.Core.Contracts.General.Labels> RetrieveAll(int companyId, string filter, int beginIndex, int endIndex, List<DataFactory.Contracts.Parameter> parameters = null, IDbConnection connection = null)
        {
            List<Architect.API.Core.Contracts.General.Labels> result = new List<Architect.API.Core.Contracts.General.Labels>();
            if (beginIndex == 0)
            {
                beginIndex = 1;
            }
            if (endIndex == 0 || endIndex < beginIndex)
            {
                endIndex = int.MaxValue;
            }
            Database.Select("SELECT * FROM (" +
                            "SELECT Id, Labels.CompanyId, EntityType, EntityId, LabelId, Labels.UpdateUserCode, um.FirstName || ' ' || um.LastName AS UpdateUserName, Labels.UpdateDate " +
                                   ", ROW_NUMBER() OVER (ORDER BY Labels.Id DESC) RowNumber " +
                              "FROM Labels LEFT JOIN UserMember um ON um.UserId = Labels.UpdateUserCode " +
                             "WHERE Labels.CompanyId=:CompanyId" + filter +
                               ") WHERE RowNumber BETWEEN :beginIndex AND :endIndex")
                        .AddParameter("CompanyId", DbType.Decimal, 5, companyId)
                        .AddParameter(parameters)
                        .AddParameter("beginIndex", DbType.Decimal, 9, beginIndex)
                        .AddParameter("endIndex", DbType.Decimal, 9, endIndex)
                        .Query(connection, "Research", new Action<System.Data.IDataReader>((reader) =>
                        {
                            result.Add(DataReaderToLabels(reader));
                        }));
            return result;
        }

        /// <summary>
        /// Genera complemento del query para habilitar el filtro por columnas establecidas.
        /// </summary>
        /// <param name="filter">Permite filtrar la lista de registros.</param>
        /// <param name="where">Indica que el complement debe incluir el WHERE del comando.</param>
        /// <returns>Complemento del query</returns>
        public static string FilterBuilder(string filter, bool includeWhere = true)
        {
            string result = string.Empty;


            return result;
        }

        /// <summary>
        /// Último valor asignado a clave unica de la tabla Labels.
        /// </summary>
        /// <param name="connection">Instancia de una conexión compartida</param>
        /// <returns>Último valor asignado.</returns>
        public static int RetrieveLastKey(IDbConnection connection = null)
        {

            return (int)Database.Select("SELECT NVL(MAX(Id),0) " +
                                     "FROM Labels")
                                .QueryScalar<Decimal>(connection, "Research");
        }

        /// <summary>
        /// Actualiza un registro en la tabla Labels por medio de su clave primaria.
        /// </summary>
        /// <param name="labelsItem">Instancia de Labels</param>
        /// <param name="connection">Instancia de una conexión compartida</param>
        /// <returns>Cantidad de registros actualizados.</returns>
        public static int Update(Architect.API.Core.Contracts.General.Labels labelsItem, IDbConnection connection = null)
        {
            if (labelsItem.UpdateDate.IsEmpty())
            {
                labelsItem.UpdateDate = DateTime.Now;
            }
            return Database.Update("UPDATE Labels " +
                                      "SET CompanyId=:CompanyId, EntityType=:EntityType, EntityId=:EntityId, LabelId=:LabelId, UpdateUserCode=:UpdateUserCode, UpdateDate=:UpdateDate " +
                                    "WHERE Id=:Id")
                                .AddParameter("CompanyId", DbType.Decimal, 5, labelsItem.CompanyId)
                                .AddParameter("EntityType", DbType.Decimal, 5, labelsItem.EntityType)
                                .AddParameter("EntityId", DbType.Decimal, 18, labelsItem.EntityId)
                                .AddParameter("LabelId", DbType.Decimal, 5, labelsItem.LabelId)
                                .AddParameter("UpdateUserCode", DbType.Decimal, 9, labelsItem.UpdateUserCode)
                                .AddParameter("UpdateDate", DbType.DateTime, 0, labelsItem.UpdateDate)
                                .AddParameter("Id", DbType.Decimal, 9, labelsItem.Id)
                                .Execute(connection, "Research");
        }

        /// <summary>
        /// Actualiza una lista de registros en la tabla Labels por medio de su clave primaria.
        /// </summary>
        /// <remarks>Complemento para procesamiento masivo</remarks>
        /// <param name="labelsItems">Lista de instancia de Labels</param>
        /// <param name="connection">Instancia de una conexión compartida</param>
        /// <returns>Lista con el resultado de la creación de cada instancia.</returns>
        public static List<int> Update(List<Architect.API.Core.Contracts.General.Labels> labelsItems, IDbConnection connection = null)
        {
            List<int> result = new List<int>();
            bool local = false;

            if (connection == null)
            {
                connection = Architect.DataFactory.Database.OpenConnection("Research");
                local = true;
            }
            foreach (Architect.API.Core.Contracts.General.Labels item in labelsItems)
            {
                result.Add(Update(item, connection));
            }
            if (local)
            {
                connection.Close();
            }
            return result;
        }

        /// <summary>
        /// Elimina un registro en la tabla Labels por medio de su clave primaria.
        /// </summary>
        /// <param name="id">Identificación única del registro.</param>
        /// <param name="companyId">Identificación de la compañia propietaria.</param>
        /// <param name="connection">Instancia de una conexión compartida</param>
        /// <returns>Cantidad de registros eliminados.</returns>
        public static int Delete(int id, int companyId, IDbConnection connection = null)
        {
            return Database.Delete("DELETE FROM Labels " +
                                    "WHERE Id=:Id AND CompanyId=:CompanyId")
                                .AddParameter("Id", DbType.Decimal, 9, id)
                                .AddParameter("CompanyId", DbType.Decimal, 5, companyId)
                                .Execute(connection, "Research");
        }

        /// <summary>
        /// Actualiza una lista de registros en la tabla Labels por medio de su clave primaria.
        /// </summary>
        /// <remarks>Complemento para procesamiento masivo</remarks>
        /// <param name="idList">Lista de Identificación única del registro.</param>
        /// <param name="companyId">Identificación de la compañia propietaria.</param>
        /// <param name="connection">Instancia de una conexión compartida</param>
        /// <returns>Lista con el resultado de la creación de cada instancia.</returns>
        public static List<int> Delete(List<int> idList, int companyId, IDbConnection connection = null)
        {
            List<int> result = new List<int>();
            bool local = false;

            if (connection == null)
            {
                connection = Architect.DataFactory.Database.OpenConnection("Research");
                local = true;
            }
            foreach (int item in idList)
            {
                result.Add(Delete(item, companyId, connection));
            }
            if (local)
            {
                connection.Close();
            }
            return result;
        }

        /// <summary>
        /// Recupera la cantidad de registros existentes en la tabla Labels por medio de su clave primaria.
        /// </summary>
        /// <param name="id">Identificación única del registro.</param>
        /// <param name="companyId">Identificación de la compañia propietaria.</param>
        /// <param name="connection">Instancia de una conexión compartida</param>
        /// <returns>Cantidad de registros encontrados.</returns>
        public static int Count(int id, int companyId, IDbConnection connection = null)
        {

            return (int)Database.Select("SELECT COUNT(Id) " +
                                          "FROM Labels " +
                                         "WHERE Id=:Id AND CompanyId=:CompanyId")
                               .AddParameter("Id", DbType.Decimal, 9, id)
                               .AddParameter("CompanyId", DbType.Decimal, 5, companyId)
                               .QueryScalar<Decimal>(connection, "Research");
        }

        /// <summary>
        /// Recupera la cantidad de registros existentes en la tabla Labels que cumplen con el filtro.
        /// </summary>
        /// <param name="filter">Permite filtrar la lista de registros</param>
        /// <param name="parameters">Lista de parametros para complemento.</param>
        /// <param name="connection">Instancia de una conexión compartida</param>
        /// <returns>Cantidad de registros encontrados.</returns>
        public static int Count(string filter, List<DataFactory.Contracts.Parameter> parameters = null, IDbConnection connection = null)
        {
            return (int)Database.Select("SELECT COUNT(Id) " +
                                          "FROM Labels " + filter)
                               .AddParameter(parameters)
                               .QueryScalar<Decimal>(connection, "Research");
        }

        /// <summary>
        /// Convierte una fila de un 'DataReader' a un clase de tipo 'Labels'.
        /// </summary>
        /// <param name="reader">DataReader</param>
        /// <param name="item">Instancia pre creada</param>
        /// <returns>Instancia con la información de la fila del 'DataReader'</returns>
        public static Architect.API.Core.Contracts.General.Labels DataReaderToLabels(System.Data.IDataReader reader, Architect.API.Core.Contracts.General.Labels item = null)
        {
            if (item == null)
            {
                item = new Architect.API.Core.Contracts.General.Labels();
            }
            item.Id = reader.IntegerValue("Id");
            item.CompanyId = reader.IntegerValue("CompanyId");
            item.EntityType = reader.IntegerValue("EntityType");
            item.EntityId = reader.Integer64Value("EntityId");
            item.LabelId = reader.IntegerValue("LabelId");
            item.UpdateUserCode = reader.IntegerValue("UpdateUserCode");
            item.UpdateUserName = reader.StringValue("UpdateUserName");
            item.UpdateDate = reader.DateTimeValue("UpdateDate");
            return item;
        }

    }

}

File: Core/Architect.API.Core.DataAccess/_Generated/Labels.vb
Imports Architect.Common.Proxy

Namespace Architect.API.Core.DataAccess.General

    ''' <summary>
    ''' 
    ''' </summary>
    Partial Public NotInheritable Class Labels

        ''' <summary>
        ''' Crea un registro en la tabla Labels.
        ''' </summary>
        ''' <param name="labelsItem">Instancia de Labels</param>
        ''' <returns>Instancia de Labels</returns>
        Public Shared Function Create(labelsItem As Architect.API.Core.Contracts.General.Labels) As Integer
            Dim rows As Integer = 0
            With New DataManagerFactory("INSERT INTO Labels " &
                                              "(Id, Sequence, LabelId, EntityType, EntityId, CompanyId, UpdateUserCode, UpdateDate) " &
                                        "VALUES(@:Id, @:Sequence, @:LabelId, @:EntityType, @:EntityId, @:CompanyId, @:UpdateUserCode, @:UpdateDate)", "Labels", "Research")
                .AddParameter("Id", DbType.Decimal, 9, labelsItem.Id.IsEmpty(), labelsItem.Id)
                '.AddParameter("Sequence", DbType.Decimal, 9, labelsItem.Sequence.IsEmpty(), labelsItem.Sequence)
                .AddParameter("LabelId", DbType.Decimal, 5, labelsItem.LabelId.IsEmpty(), labelsItem.LabelId)
                .AddParameter("EntityType", DbType.Decimal, 5, labelsItem.EntityType.IsEmpty(), labelsItem.EntityType)
                .AddParameter("EntityId", DbType.Decimal, 9, labelsItem.EntityId.IsEmpty(), labelsItem.EntityId)
                .AddParameter("CompanyId", DbType.Decimal, 5, labelsItem.CompanyId.IsEmpty(), labelsItem.CompanyId)
                .AddParameter("UpdateUserCode", DbType.Decimal, 9, labelsItem.UpdateUserCode.IsEmpty(), labelsItem.UpdateUserCode)
                .AddParameter("UpdateDate", DbType.DateTime, 0, labelsItem.UpdateDate.IsEmpty(), labelsItem.UpdateDate)
                rows = .CommandExecute()
            End With
            Return rows
        End Function

        ''' <summary>
        ''' Recupera un registro en la tabla Labels por medio de su clave primaria.
        ''' </summary>
        ''' <param name="id"></param>
        ''' <param name="sequence"></param>
        ''' <returns>Instancia de Labels</returns>
        Public Shared Function Retrieve(id As Integer, sequence As Integer) As Architect.API.Core.Contracts.General.Labels
            Dim rows As Data.DataTable = Nothing
            Dim result As Architect.API.Core.Contracts.General.Labels = Nothing
            With New DataManagerFactory("SELECT Id, Sequence, LabelId, EntityType, EntityId, Labels.CompanyId, Labels.UpdateUserCode, um.FirstName || ' ' || um.LastName AS UpdateUserName, Labels.UpdateDate " &
                                          "FROM Labels LEFT JOIN UserMember um ON um.UserId = Labels.UpdateUserCode " &
                                         "WHERE Id=@:Id AND Sequence=@:Sequence", "Labels", "Research")
                .AddParameter("Id", DbType.Decimal, 9, False, id)
                .AddParameter("Sequence", DbType.Decimal, 9, False, sequence)
                rows = .QueryExecuteToTable(True)
                If rows.IsNotEmpty AndAlso rows.Rows.Count <> 0 Then
                    result = DataRowMapper(rows.Rows(0))
                End If
            End With
            Return result
        End Function

        ''' <summary>
        ''' Recupera una lista de registros en la tabla Labels.
        ''' </summary>
        ''' <returns>Lista de instancias de Labels</returns>
        Public Shared Function RetrieveAll() As List(Of Architect.API.Core.Contracts.General.Labels)
            Dim rows As Data.DataTable = Nothing
            Dim result As New List(Of Architect.API.Core.Contracts.General.Labels)
            With New DataManagerFactory("SELECT Id, Sequence, LabelId, EntityType, EntityId, Labels.CompanyId, Labels.UpdateUserCode, um.FirstName || ' ' || um.LastName AS UpdateUserName, Labels.UpdateDate " &
                                          "FROM Labels LEFT JOIN UserMember um ON um.UserId = Labels.UpdateUserCode", "Labels", "Research")
                rows = .QueryExecuteToTable(True)
                If rows.IsNotEmpty AndAlso rows.Rows.Count <> 0 Then
                    For Each itemRow As DataRow In rows.Rows
                        result.Add(DataRowMapper(itemRow))
                    Next
                End If
            End With
            Return result
        End Function

        ''' <summary>
        ''' Verifica si existe un registro en la tabla Labels por medio de su clave primaria.
        ''' </summary>
        ''' <param name="id"></param>
        ''' <returns>Verdadero si existe el registro, en caso contrario falso.</returns>
        Public Shared Function RetrieveLastKey(id As Integer) As Integer
            Dim rows As Integer = 0

            With New DataManagerFactory("SELECT NVL(MAX(Sequence),0) " &
                                          "FROM Labels " &
                                         "WHERE Id=@:Id", "Labels", "Research")
                .AddParameter("Id", DbType.Decimal, 9, False, id)
                rows = .QueryExecuteScalarToInteger
            End With
            Return rows
        End Function

        ''' <summary>
        ''' Actualiza un registro en la tabla Labels por medio de su clave primaria.
        ''' </summary>
        ''' <param name="labelsItem">Instancia de Labels</param>
        ''' <returns>Cantidad de registros actualizados.</returns>
        Public Shared Function Update(labelsItem As Architect.API.Core.Contracts.General.Labels) As Integer
            Dim rows As Integer = 0
            With New DataManagerFactory("UPDATE Labels " &
                                           "SET LabelId=@:LabelId, EntityType=@:EntityType, EntityId=@:EntityId, CompanyId=@:CompanyId, UpdateUserCode=@:UpdateUserCode, UpdateDate=@:UpdateDate " &
                                         "WHERE Id=@:Id AND Sequence=@:Sequence", "Labels", "Research")
                .AddParameter("LabelId", DbType.Decimal, 5, labelsItem.LabelId.IsEmpty(), labelsItem.LabelId)
                .AddParameter("EntityType", DbType.Decimal, 5, labelsItem.EntityType.IsEmpty(), labelsItem.EntityType)
                .AddParameter("EntityId", DbType.Decimal, 9, labelsItem.EntityId.IsEmpty(), labelsItem.EntityId)
                .AddParameter("CompanyId", DbType.Decimal, 5, labelsItem.CompanyId.IsEmpty(), labelsItem.CompanyId)
                .AddParameter("UpdateUserCode", DbType.Decimal, 9, labelsItem.UpdateUserCode.IsEmpty(), labelsItem.UpdateUserCode)
                .AddParameter("UpdateDate", DbType.DateTime, 0, labelsItem.UpdateDate.IsEmpty(), labelsItem.UpdateDate)
                .AddParameter("Id", DbType.Decimal, 9, False, labelsItem.Id)
                '.AddParameter("Sequence", DbType.Decimal, 9, False, labelsItem.Sequence)
                rows = .CommandExecute()
            End With
            Return rows
        End Function

        ''' <summary>
        ''' Elimina un registro en la tabla Labels por medio de su clave primaria.
        ''' </summary>
        ''' <param name="id"></param>
        ''' <param name="sequence"></param>
        ''' <returns>Cantidad de registros eliminados.</returns>
        Public Shared Function Delete(id As Integer, sequence As Integer) As Integer
            Dim rows As Integer = 0
            With New DataManagerFactory("DELETE FROM Labels " &
                                         "WHERE Id=@:Id AND Sequence=@:Sequence", "Labels", "Research")
                .AddParameter("Id", DbType.Decimal, 9, False, id)
                .AddParameter("Sequence", DbType.Decimal, 9, False, sequence)
                rows = .CommandExecute()
            End With
            Return rows
        End Function

        ''' <summary>
        ''' Recupera la cantidad de registros existentes en la tabla Labels por medio de su clave primaria.
        ''' </summary>
        ''' <param name="id"></param>
        ''' <param name="sequence"></param>
        ''' <returns>Cantidad de registros encontrados.</returns>
        Public Shared Function Count(id As Integer, sequence As Integer) As Integer
            Dim rows As Integer = 0

            With New DataManagerFactory("SELECT COUNT(Id) " &
                                          "FROM Labels " &
                                         "WHERE Id=@:Id AND Sequence=@:Sequence", "Labels", "Research")
                .AddParameter("Id", DbType.Decimal, 9, False, id)
                .AddParameter("Sequence", DbType.Decimal, 9, False, sequence)
                rows = .QueryExecuteScalarToInteger
            End With
            Return rows
        End Function

        ''' <summary>
        ''' Recupera una lista de registros en la tabla Labels.
        ''' </summary>
        ''' <param name="entityid"></param>
        ''' <param name="entitytype"></param>
        ''' <param name="companyid"></param>
        ''' <returns>Lista de instancias de Labels</returns>
        Public Shared Function RetrieveByEntityIdEntityTypeCompanyId(entityid As Integer, entitytype As Integer, companyid As Integer) As List(Of Architect.API.Core.Contracts.General.Labels)
            Dim rows As Data.DataTable = Nothing
            Dim result As New List(Of Architect.API.Core.Contracts.General.Labels)
            With New DataManagerFactory("Select Id, Sequence, LabelId, EntityType, EntityId, Labels.CompanyId, Labels.UpdateUserCode, um.FirstName || ' ' || um.LastName AS UpdateUserName, Labels.UpdateDate " &
                                          "FROM Labels LEFT JOIN UserMember um ON um.UserId = Labels.UpdateUserCode " &
                                         "WHERE EntityId=@:EntityId AND EntityType=@:EntityType AND CompanyId=@:CompanyId", "Labels", "Research")
                .AddParameter("EntityId", DbType.Decimal, 9, False, entityid)
                .AddParameter("EntityType", DbType.Decimal, 5, False, entitytype)
                .AddParameter("CompanyId", DbType.Decimal, 5, False, companyid)
                rows = .QueryExecuteToTable(True)
                If rows.IsNotEmpty AndAlso rows.Rows.Count <> 0 Then
                    For Each itemRow As DataRow In rows.Rows
                        result.Add(DataRowMapper(itemRow))
                    Next
                End If
            End With
            Return result
        End Function

        ''' <summary>
        ''' Elimina un registro en la tabla Labels por medio de su clave primaria.
        ''' </summary>
        ''' <param name="entityid"></param>
        ''' <param name="entitytype"></param>
        ''' <param name="companyid"></param>
        ''' <returns>Cantidad de registros eliminados.</returns>
        Public Shared Function DeleteByEntityIdEntityTypeCompanyId(entityid As Integer, entitytype As Integer, companyid As Integer) As Integer
            Dim rows As Integer = 0
            With New DataManagerFactory("DELETE FROM Labels " &
                                         "WHERE EntityId=@:EntityId AND EntityType=@:EntityType AND CompanyId=@:CompanyId", "Labels", "Research")
                .AddParameter("EntityId", DbType.Decimal, 9, False, entityid)
                .AddParameter("EntityType", DbType.Decimal, 5, False, entitytype)
                .AddParameter("CompanyId", DbType.Decimal, 5, False, companyid)
                rows = .CommandExecute()
            End With
            Return rows
        End Function

#Region "Helper"

        ''' <summary>
        ''' Permite hacer el mapeo entre una fila de un datatable hacia una instancia de Labels.
        ''' </summary>
        ''' <returns>Instancia de Labels</returns>
        Private Shared Function DataRowMapper(itemRow As DataRow) As Architect.API.Core.Contracts.General.Labels
            Return New Architect.API.Core.Contracts.General.Labels With {.Id = itemRow.NumericValue("Id"), .LabelId = itemRow.NumericValue("LabelId"), .EntityType = itemRow.NumericValue("EntityType"), .EntityId = itemRow.NumericValue("EntityId"), .CompanyId = itemRow.NumericValue("CompanyId"), .UpdateUserCode = itemRow.NumericValue("UpdateUserCode"), .UpdateUserName = itemRow.StringValue("UpdateUserName"), .UpdateDate = itemRow.DateTimeValue("UpdateDate")}
        End Function

#End Region

    End Class

End Namespace

